docsite/docs/Docs/fal-cli/environments.md

```diff
@@ -0,0 +1,298 @@
+> specifying the `FAL_CONDA_HOME` environment variable. E.g. if you have a
+> `miniconda` installation under `~/Downloads`, you can start fal with
+> `FAL_CONDA_HOME=~/Downloads/miniconda3/bin fal flow run`.
+
+Fal currently uses the global `~/.condarc` file when looking for channels to
+install packages (or otherwise defaulting to the `conda` executable's bundled
+search channels), so if you want to use packages from `conda-forge` (or any
+other non-default channels), you can configure them with the following commands:
+
+```console
+$ conda config --add channels conda-forge
+$ conda config --set channel_priority strict
+```
+
+## Specifying environments
+
+You can specify environments for your hooks and models under your schema file
+(`schema.yml`). There are currently two scopes to configure the environment
+option, one is in the model-level (which all the hooks would automatically
+inherit from) and the other one is in the hook-level (so only the selected hook
+would run in the specified environment, and nothing else would be affected by
+it).
+
+### Changing a hook's environment
+
+If you only want to run a hook in a specific environment, you can define the
+`environment` attribute for that hook and Fal will ensure that the hook will be
+executed under the specified environment.
+
+```yml
+version: 2
+
+models:
+  - name: reorder_alphabetically
+    meta:
+      fal:
+        pre-hook:
+          - print_enter_banner.py
+
+        post-hook:
+          - print_exit_banner.py
+          - path: send_slack_alert.py
+            environment: alerts
+```
+
+In the example above, both the model `reoder_alphabetically` and the banner
+printing scripts (`print_enter_banner.py` / `print_exit_banner.py`) will use
+your computer's local environment (think of it as the Python environment Fal
+itself is installed on) since they might not depend on any external Python
+packages. But the `send_slack_alert.py` hook will use the `alerts` environment
+that is defined under `fal_project.yml` which contains `slack-sdk` as a
+dependency. So no matter where this runs (your computer, your co-worker's
+computer or your project's CI environment), Fal will create a new virtual
+environment and install all the necessary requirements that needs to be
+available for `send_slack_alert.py` to work properly and a send a slack
+notification.
+
+### Changing a model's environment (with all the underlying hooks)
+
+If you have a Fal Python model and want it to run on an isolated environment,
+you can change the model-level environment which would then make Fal run your
+Python model (as well as all the pre-hooks/post-hooks, unless you override it)
+on the specified environment.
+
+```yml
+version: 2
+
+models:
+  - name: load_data_from_s3
+    meta:
+      fal:
+        environment: interact-cloud
+        pre-hook:
+          - path: change_s3_permissions.py
+            with:
+              can_dbt_read: yes
```

Comment:
`dbt_has_environment_access` might be better?

Reply:
Made it a bit more clear, thanks!